suspendmanager.lua

```diff
@@ -151,14 +151,23 @@ local ERASABLE_DESIGNATION = {
 }
 
 --- Job types that impact suspendmanager
-local FILTER_JOB_TYPES = {
-    [df.job_type.CarveTrack]=true,
-    [df.job_type.ConstructBuilding]=true,
-    [df.job_type.DestroyBuilding]=true,
-    [df.job_type.DetailFloor]=true,
-    [df.job_type.Dig]=true,
-    [df.job_type.DigChannel]=true,
-    [df.job_type.SmoothFloor]=true,
+--- Any completed pathable job can impact suspendmanager by allowing or disallowing
+--- access to construction job
+local FILTER_JOB_TYPES = utils.invert{
+    df.job_type.CarveRamp,
+    df.job_type.CarveTrack,
+    df.job_type.CarveUpDownStaircase,
+    df.job_type.CarveUpwardStaircase,
+    df.job_type.CarveDownwardStaircase,
+    df.job_type.ConstructBuilding,
+    df.job_type.DestroyBuilding,
+    df.job_type.DetailFloor,
```

Comment:
Why are DetailFloor and SmoothFloor in this list? How do they affect access?

Reply:
I've expanded the comment, it's to properly update the ERASE_DESIGNATION suspension reason, a smooth/carving/detailing job will require any job on top of it to be suspended to avoid being erased.

In gif form, this chair is about to be built, but I add at the last second a smoothing job. Completing the chair would erase the smoothing job, so it suspends the chair. If SmoothFloor was not part of the filter, then the suspension would not have happened immediately
![suspendonsmooth](https://github.com/DFHack/scripts/assets/630159/8f914e56-c906-4bc0-9d51-eda5b1f1718a)